File: 3. Big O/1.finding_nemo.js
const names = ['nemo'];
const large = new Array(100000).fill('nemo')

function findNemo (array) {
   t0 = performance.now();
   for (let i = 0; i < array.length; i++) {
      console.log("Found Nemo!");
      break
   }
   t1 = performance.now();
   console.log(`findNemo function took ${t1-t0} ms`)

}

findNemo(large)

File: 3. Big O/1.finding_nemo.py
# Python - 3.10
# Topic - DS & Algorithm
# program - A simple function that finds nemo
import time 

names1 = ['nemo']
names100 = ['nemo'] * 100
names1000 = ['nemo'] * 1000
names10000 = ['nemo'] * 10000
names100000 = ['nemo'] * 100000

# print(names)
def findNemo(array):
   start = time.time()
   for i in array:
      if i == "nemo":
         print("Found Nemo!")
   end = time.time()
   print(f'findNemo function took {end-start} ms')

findNemo(names10000)